ssl/ssl_lib.c

```diff
@@ -5180,7 +5178,11 @@ int SSL_client_version(const SSL *s)
 {
     const SSL_CONNECTION *sc = SSL_CONNECTION_FROM_CONST_SSL(s);
 
-    /* TODO(QUIC): Do we want to report QUIC version this way instead? */
+#ifndef OPENSSL_NO_QUIC
+    /* We only support QUICv1 - so if its QUIC its QUICv1 */
+    if (s->type == SSL_TYPE_QUIC_CONNECTION || s->type == SSL_TYPE_QUIC_XSO)
+        return OSSL_QUIC1_VERSION;
+#endif
```

Comment:
I disagree. I do not think using SSL_get_version/SSL_version/SSL_client_version() to represent the handshake TLS version would be a good idea. IMO we should add a new APIs for that once it is needed, i.e. when it would be possible to use some future TLS versions for the handshake.

Whether we return 0 or OSSL_QUIC1_VERSION from SSL_client_version() is debatable but IMO it is still better to return OSSL_QUIC1_VERSION here.

I know there is the type mismatch between the QUIC version being an unsigned 32 bit number and the int type and also a possibility for clash with TLS version numbers but with the current version numbers there is no such problem. We also never documented that OSSL_QUIC1_VERSION is equal to the number as defined by the RFC so we might even just use 2, 3, 4,... for QUIC2, QUIC3, etc. regardless of the version number in the protocol.

We could also keep returning OSSL_QUIC1_VERSION for QUIC regardless of the version. I do not think this really complicates any future changes or additions of version handling in the QUIC SSL API.

Reply:
I've also changed the SSL_client_version() documentation to reflect this change.

Reply:
In that case, perhaps we can change `OSSL_QUIC1_VERSION` to something different to the wire version number to prevent accidental dependency by applications?